packages/@lwc/synthetic-shadow/src/faux-shadow/events.ts

```diff
@@ -173,73 +40,40 @@ function getEventMap(elm: EventTarget): ListenerMap {
 }
 
 const shadowRootEventListenerMap: WeakMap<EventListener, WrappedListener> = new WeakMap();
+// Using a WeakMap instead of a WeakSet because this one works in IE11 :(
+const eventsComingFromShadowRoot: WeakMap<Event, 1> = new WeakMap();
+
+function isEventComingFromShadowRoot(event: Event): boolean {
+    return eventsComingFromShadowRoot.has(event);
+}
+
+export function setEventFromShadowRoot(event: Event) {
+    eventsComingFromShadowRoot.set(event, 1);
+}
 
 function getWrappedShadowRootListener(
     sr: SyntheticShadowRootInterface,
     listener: EventListener
 ): WrappedListener {
-    if (!isFunction(listener)) {
```

Comment:
We are relaxing this condition. But we are not account for a listener which implements the [EventListener interface](https://developer.mozilla.org/en-US/docs/Web/API/EventListener).

Reply:
good catch! I thought we were triggering this from another patch, but that's not the case... will look at solving this.